### PR TITLE
chore: adding useScreenSize hook

### PR DESCRIPTION
**Pull request by JackShort**
* this is implementing new useScreenHook to deprecate useIsTablet and useIsMobile

### CODE REVIEW

File: src/nft/hooks/useIsMobile.ts [hunk 1]
```diff
@@ -1,28 +1,9 @@
-import { breakpoints } from 'nft/css/sprinkles.css'
-import { useEffect, useState } from 'react'
-
-const isClient = typeof window !== 'undefined'
-
-export function getIsMobile() {
-  return isClient ? window.innerWidth < breakpoints.sm : false
-}
+// @deprecated in favor of useScreenSize
```
Comment (cbachmeier) [inline src/nft/hooks/useIsMobile.ts:1]: Can you give the same treatment to `src/nft/hooks/useIsTablet.ts`?

File: src/nft/hooks/useIsMobile.ts [hunk 2]
```diff
@@ -1,28 +1,9 @@
...
-    if (isClient) {
-      window.addEventListener('resize', handleResize)
-      return () => {
-        window.removeEventListener('resize', handleResize)
-      }
-    }
-    return undefined
-  }, [])
+  const isScreenSize = useScreenSize()
+  const isMobile = !isScreenSize['sm']
```
Comment (cbachmeier) [inline src/nft/hooks/useIsMobile.ts:6]: The syntax isMobile = **not** isScreenSize['sm'] seems backwards? I would assume using the hook isMobile should equal isScreenSize['sm']

Comment (JackShort) [reply src/nft/hooks/useIsMobile.ts:6]: resolved over slack

File: src/nft/utils/scrollToTop.ts [hunk 3]
```diff
@@ -1,4 +1,9 @@
-import { getIsMobile } from 'nft/hooks'
+import { breakpoints } from 'nft/css/sprinkles.css'
+const isClient = typeof window !== 'undefined'
+
+function getIsMobile() {
+  return isClient ? window.innerWidth < breakpoints.sm : false
+}
```
[the remark this answers is not in the document]
Comment (JackShort) [reply src/nft/utils/scrollToTop.ts:6]: because this is a util function and not a hook. I don't think it should instantiate a hook in it.

**Review by cbachmeier — APPROVED** [review]
LGTM. Can you just add some comments to useScreenSize of what to enter and what to expect to have returned sos that someone can quickly pick it up and use it?

**Pull request merged**